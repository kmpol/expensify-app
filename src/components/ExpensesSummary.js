import React from 'react'
import { connect } from 'react-redux';
import selectedExpenses from '../selectors/expenses'
import selectedExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export class ExpensesSummary extends React.Component {

    render() {
        const grammarExpenseDisplay = this.props.expenseCount === 1 ? "expense" : "expenses"
        return (
            <div>
                <p>Viewing {this.props.expenseCount} {grammarExpenseDisplay} with total: ${numeral(this.props.expensesInTotal / 100).format('0.00')}</p>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectedExpenses(state.expenses, state.filters)

    return {
        expenseCount: visibleExpenses.length,
        expensesInTotal: selectedExpensesTotal(visibleExpenses)
    }
}
export default connect(mapStateToProps)(ExpensesSummary)