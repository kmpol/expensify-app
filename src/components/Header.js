import { NavLink } from 'react-router-dom'
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux'

export const Header = ({ startLogout }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink activeClassName="is-active" to="/dashboard">Home page</NavLink>
        <NavLink activeClassName="is-active" to="/create">Create expense</NavLink>
        <NavLink activeClassName="is-active" to="/help">Help page</NavLink>
        <button onClick={startLogout()}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout)
    }
}

export default connect(undefined, mapDispatchToProps)(Header)