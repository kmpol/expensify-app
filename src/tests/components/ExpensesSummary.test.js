import { shallow, configure } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() });

test('Should render singular expense summary', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesInTotal={1} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})

test('Should render multiple expenses summary', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={5} expensesInTotal={500000} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})