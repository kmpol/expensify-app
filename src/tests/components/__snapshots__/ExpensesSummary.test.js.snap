// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render multiple expenses summary 1`] = `
<div>
  <p>
    Viewing 
    5
     
    expenses
     with total: $
    5000.00
  </p>
</div>
`;

exports[`Should render singular expense summary 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     with total: $
    0.01
  </p>
</div>
`;
